includes/admin/class-amp-post-meta-box.php

```diff
@@ -292,6 +295,46 @@ function () {
 		}
 	}
 
+	/**
+	 * Returns the featured image dimensions after filtering
+	 * the height and width
+	 *
+	 * @return int[]
+	 */
+	public static function get_featured_image_dimensions() {
+		$default_width  = 1200;
+		$default_height = 675;
+
+		$dimensions = [
+			/**
+			 * Filters the minimum height required for a featured image.
+			 *
+			 * @since 2.0.9
+			 *
+			 * @param int $featured_image_minimum_height The minimum height of the image,
+			 *                                           defaults to 675.
+			 */
+			'featuredImageMinimumHeight' => (int) apply_filters( 'amp_featured_image_minimum_height', $default_height ),
+
+			/**
+			 * Filters the minimum width required for a featured image.
+			 *
+			 * @since 2.0.9
+			 *
+			 * @param int $featured_image_minimum_height The minimum width of the image,
+			 *                                           defaults to 1200.
+			 */
+			'featuredImageMinimumWidth'  => (int) apply_filters( 'amp_featured_image_minimum_width', $default_width ),
+		];
+		if ( $dimensions['featuredImageMinimumHeight'] <= 0 ) {
+			$dimensions['featuredImageMinimumHeight'] = $default_height;
+		}
+		if ( $dimensions['featuredImageMinimumWidth'] <= 0 ) {
+			$dimensions['featuredImageMinimumWidth'] = $default_width;
+		}
```

Comment:
Something that comes to mind: would there be value to be able to filter `amp_featured_image_minimum_height` and/or `amp_featured_image_minimum_width` to be `-1` or `false` in order to disable constraints in one dimension, or to eliminate the constraints altogether if they are both filtered to be `-1` or `false`? I realize this contradicts what I suggested where that method always has to return valid integer dimensions.  In the case of the value being less than or equal to `0`, then the value of `false` could be supplied instead of the defaults. Then the JS would have to account for the values being either integers or `false` and update the UI accordingly. 

Would this be useful? 

Perhaps a site owner would like to disable the featured image dimension constraints altogether, and this would provide them a way to do this. cc @pierlon @johnwatkins0

Reply:
In other words, the question is whether we could actually allow a site author to do:

```php
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

or

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
```

or

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

Reply:
I'd say it's highly unlikely a site owner would want to disable constraints for one dimension. If the dimensions are filtered to a height & width of `0`, then I'd consider the constraints to be disabled.

Reply:
You're right. In any case, it seems unsymmetric to allow filtering _either_ of the dimensions to disable the constraints. I've pushed up what I had in mind.

If I do:

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
```

I get:

Media Library | Document Sidebar | Pre-publish Check
--------------|--------------------|-------------------
![image](https://user-images.githubusercontent.com/134745/102274710-28e73b00-3ed9-11eb-8773-16f1c86cbf42.png) | ![image](https://user-images.githubusercontent.com/134745/102274664-1a008880-3ed9-11eb-9d50-45edd4212af2.png) | ![image](https://user-images.githubusercontent.com/134745/102274727-313f7600-3ed9-11eb-8f6d-62b7ed35a77a.png)

 But if I instead do:

```php
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

I get:

Media Library | Document Sidebar | Pre-publish Check
--------------|--------------------|-------------------
![image](https://user-images.githubusercontent.com/134745/102274789-487e6380-3ed9-11eb-9495-81230744382c.png) | ![image](https://user-images.githubusercontent.com/134745/102274836-5a600680-3ed9-11eb-8ade-18d2af6b02e1.png) | ![image](https://user-images.githubusercontent.com/134745/102274843-5fbd5100-3ed9-11eb-9c47-c56ec50a79c6.png)

And if I do:

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

I get:

Media Library | Document Sidebar | Pre-publish Check
--------------|--------------------|-------------------
![image](https://user-images.githubusercontent.com/134745/102274904-7663a800-3ed9-11eb-8a9e-ba02ba6eab8d.png) | ![image](https://user-images.githubusercontent.com/134745/102274919-7c598900-3ed9-11eb-8b9d-57369db93a94.png) | ![image](https://user-images.githubusercontent.com/134745/102274936-824f6a00-3ed9-11eb-8192-ee455efa6c10.png)

Reply:
@naveen17797 What do you think of this? I don't believe it impacts your use case, but would allow others to disable the checks entirely.

Reply:
@westonruter My point of view would be adding a separate filter for enabling / disabling the featured image check like

add_filter( 'amp_featured_image_check', '__return_false' );

so that we will keep the concerns separate, may be i am wrong.. 

what do you think ?

Reply:
That would work too, but since we already have the `amp_featured_image_minimum_width` and `amp_featured_image_minimum_height` filters, it seems unnecessary to add yet another filter if those two can have the same effect by filtering the values to be `0`.

Reply:
> it seems unnecessary to add yet another filter

yes, so we can disable the check if the height or width value is zero ?

Reply:
If _both_ are filtered to be zero (or `false`):

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

Reply:
ok, should i need to open a separate issue for this ?

Reply:
For what? I implemented this already in this PR. See my screenshots above: https://github.com/ampproject/amp-wp/pull/5647#discussion_r543700197.

Reply:
i think this needs to be conditional ( only add the filter when the check needs to be done ? ).

`addFilter( 'editor.PostFeaturedImage', 'ampEditorBlocks/withFeaturedImageNotice', withFeaturedImageNotice );`

Reply:
That's true. That logic could be omitted entirely if both are filtered to be false/zero. It's not only there, but it would also need to be for the two other contexts shown in my screenshots above. In any case, it doesn't harm anything to have the logic in place: it just doesn't do anything.